### PR TITLE
ci: adds codecov to CI pipeline and coverage badge to README

### PR DESCRIPTION
**Pull request by thawkin3**
I've added Codecov to the CI Pipeline so that the GitHub Actions workflow will run the unit tests with the coverage report and then pipe the results over to Codecov. I also added a code coverage badge to the README so that we can show off the awesome coverage.

I based my changes off of these two previous MRs in the `focus-trap-react` repo:
- https://github.com/focus-trap/focus-trap-react/pull/227
- https://github.com/focus-trap/focus-trap-react/pull/233

Closes https://github.com/focus-trap/tabbable/issues/260

<details>
<summary>PR Checklist</summary>
<br/>

__Please leave this checklist in your PR.__

- [x] Source changes maintain stated browser compatibility.
- [x] Issue being fixed is referenced.
- ~Unit test coverage added/updated.~
- ~E2E test coverage added/updated.~
- ~Typings added/updated.~
- [x] README updated (API changes, instructions, etc.).
- [x] Changes to dependencies explained.
- ~Changeset added (run `yarn changeset` locally to add one, and follow the prompts).~

</details>

### CODE REVIEW

File: package.json
Comment (maraisr) [inline package.json:70]: Rather than include this as a dep, we should _just_ install this in the CI. There zero benefit to ever running this locally. 

So can you please do a `yarn global add codecov` in the gh actions file, and just running that after the tests run. Similarity get rid of line 33 here.

**Review by maraisr — CHANGES_REQUESTED** [review]
Solid effort mate, thank you so much! Just one comment about the coverage commands.

File: package.json
Comment (thawkin3) [reply package.json:70]: Love it, thanks! I'll take care of that soon. On that note, we should probably do the same over in the `focus-trap-react` repo since it also has `codecov` listed as a dev dependency: https://github.com/focus-trap/focus-trap-react/blob/master/package.json#L77

I can make the same change over there once we get this MR approved.

File: yarn.lock
Comment (thawkin3) [inline yarn.lock:334]: FYI: This was the result after I removed the `codecov` package with `yarn remove codecov`. I checked for what packages are requiring `@babel/plugin-proposal-nullish-coalescing-operator`, and it's only `@babel/preset-env` which does in fact request `@babel/plugin-proposal-nullish-coalescing-operator@^7.13.8`.

So I believe what we see here is correct, and the fact that there were two versions of this package previously installed was unnecessary and that something had gotten out of sync with the lock file.

File: package.json
Comment (thawkin3) [reply package.json]: After a few attempts, I think I finally have this working correctly. The CI pipeline passes anyway. 😅 

These two issues helped me resolve the problem of the global yarn package not being installed properly or not being installed where it could be found:

- https://github.com/yarnpkg/yarn/issues/648
- https://github.com/actions/setup-node/issues/85

**Comment by thawkin3** [comment]
@maraisr I think this is ready for another round of review whenever you have a minute. Thanks!

**Comment by stefcameron** [comment]
👀

**Review by stefcameron — APPROVED** [review]
This looks good to me, thank you @thawkin3 ! I'll let @maraisr chime in before merging.

**Review by maraisr — APPROVED** [review]
Thanks for this one! 🦄

**Comment by thawkin3** [comment]
Thanks guys!

**Pull request merged**
